src/lsqecc/pauli_rotations/circuit.py

```diff
@@ -131,12 +131,7 @@ def get_y_free_equivalent(self) -> "PauliOpCircuit":
         y_free_circuit = PauliOpCircuit(self.qubit_num, self.name)
 
         for block in self.ops:
-            if isinstance(block, Measurement) or (
-                cast(PauliRotation, block).rotation_amount in {Fraction(1, 8), Fraction(-1, 8)}
-            ):
-                y_free_circuit.ops.extend(block.get_y_free_equivalent())
-            else:
-                y_free_circuit.add_pauli_block(copy.deepcopy(block))
+            y_free_circuit.ops.extend(block.get_y_free_equivalent())
```

Comment:
would it make sense to implement `if` block.has_y()` before calling `y_free_equivalent()`? Or move have this if check at the top of the function?

Reply:
Yeah good point. there is a check inside PauliProductOperator that scans for ys anyways.

From my preliminary data on profiling, it turns out that a major bottleneck in the compilation are deep copies so we'll have to work in fixing those. But I think we can leave this as an optimization for then if it comes up the profiling that these deep copies are taking time